### ISSUE TITLE
Cryptography clients on local mode don't support elliptic curve `secp256k1` on Java 16+

### ISSUE THREAD

**Issue by vcolin7** [Azure/azure-sdk-for-java#26689]
Oracle stopped supporting some elliptic curves [starting on Java 16](https://docs.oracle.com/en/java/javase/17/migrate/removed-tools-and-components.html#GUID-F182E075-858A-4468-9434-8FC1704E7BB7) and one of those is `secp256k1`, which needs to be supported by Key Vault Key's CryptographyClient when running on local mode only, that is, not communicating with the Key Vault service to perform cryptographic operations.

Given that we are trying to transition from supporting Java 11 to Java 17 in the Azure SDK, we need to figure out a way to support cryptographic operations that use this algorithm.

Options for achieving this are:
- Using a third-party library
- Writing our own implementation
- Allowing users to pass their own Security Providers via method overloads. (Suggested in #25570 as well for different reasons).
- Shading the code that Oracle used for this and including it with the SDK

NOTE: Tests that fail due to this on Java 17 have been disabled for the time being.

**Comment by eliquinox** [comment]
any news on this?

**Comment by hhanquez** [comment]
any news ?

**Comment by pentiak** [comment]
Any chance of getting it addressed?

**Comment by VinceBCD** [comment]
Any news ?

**Comment by vcolin7** [comment]
Hi everyone, sorry for the lack of updates, we had a few other items ahead of this in our queue. I will start working on an alternative for our Java 16+ users soon. In the meantime, you can still create and use keys with elliptic curve `secp256k1` by using the Cryptography clients in default mode, that is, letting the client delegate these tasks to the Key Vault service as opposed to doing the operations locally.

**Comment by pentiak** [comment]
@usmansaleem, looking at your mention, have you confirmed that both local and default(remote) signing modes do not work on newer JREs?

**Comment by usmansaleem** [comment]
@pentiak our code is constructing CryptographyClient is follows:
```
    return new CryptographyClientBuilder()
        .credential(tokenCredential)
        .keyIdentifier(keyId)
        .buildClient();
```
The exception is raised when we perform `cryptoClient.sign(...)` operation. I am not 100% sure how to set local mode? The default mode definitely failed in Java 17 because it does not support SECP256K1 curve anymore (though BouncyCastle does). We need SECP256K1 because our code deals in signing for Eth1 private keys.

**Comment by pentiak** [comment]
I would really like this bug to be fixed. It is the only thing holding us back from upgrading JRE & Spring.

**Comment by hhanquez** [comment]
Please fix this bug !
Or at least answer to this question !

> How do you switch CryptographyClient from local mode to default mode?

**Comment by vcolin7** [comment]
Hey folks, I'm sorry it took so long to address this. I just merged a change in PR #36657 to ensure the `CryptographyClient` goes back to use the service if it was not able to create a crypto client for local-only operation, which is the case when trying to use elliptic curve `secp256k1` in Java 16+. We'll release this change in a new version in the next few days.
